src/main/java/net/earthcomputer/clientcommands/command/AliasCommand.java

```diff
@@ -0,0 +1,174 @@
+            .then(literal("remove")
+                .then(argument("key", string())
+                    .executes(ctx -> removeAlias(getString(ctx, "key"))))));
+
+        HashMap<String, String> aliasMap = getAliases();
+
+        if (aliasMap == null || aliasMap.isEmpty()) {
+            sendError(new LiteralText("No aliases registered"));
+        } else {
+            for(String key: aliasMap.keySet()) {
+                addClientSideCommand(key);
+                dispatcher.register(literal(key)
+                        .executes(ctx -> executeAliasCommand(key, null))
+                        .then(argument("arguments", greedyString())
+                                .executes(ctx -> executeAliasCommand(key, getString(ctx, "arguments")))));
+            }
+        }
+
+    }
+    private static int executeAliasCommand(String aliasKey, String arguments) {
+
+        // TODO: add support for optional greedy text arguments(?)
+        HashMap<String, String> aliasMap = getAliases();
+        String cmd;
+        if(aliasMap.containsKey(aliasKey)) {
+            cmd = aliasMap.get(aliasKey);
+        } else {
+            sendError(new LiteralText("No such key exists"));
+            return 0;
+        }
+
+        if(arguments!=null) {
+            cmd = String.format(cmd, arguments.split(" "));
+        }
+        MinecraftClient.getInstance().player.sendChatMessage(cmd);
+
+        return 0;
+    }
+
+    private static int addAlias(String key, String command) {
+        ClientPlayNetworkHandler networkHandler = MinecraftClient.getInstance().getNetworkHandler();
+        assert networkHandler != null;
+        var dispatcher = (CommandDispatcher<ServerCommandSource>) (CommandDispatcher<?>) networkHandler.getCommandDispatcher();
+
+        HashMap<String, String> aliasMap = getAliases();
+
+        assert aliasMap != null;
+        if(aliasMap.containsKey(key)) {
+            sendError(new LiteralText("That key already exists!"));
+            return 0;
+        }
+        addClientSideCommand(key);
+
+        dispatcher.register(literal(key)
+                .executes(ctx -> executeAliasCommand(key, null))
+                .then(argument("arguments", greedyString())
+                        .executes(ctx -> executeAliasCommand(key, getString(ctx, "arguments")))));
+        aliasMap.put(key, command);
+
+        try {
+            Writer writer = new FileWriter(String.valueOf(aliasPath));
+            Gson gson = new Gson();
+
+            gson.toJson(aliasMap, writer);
+            writer.flush();
+            writer.close();
+
+            sendFeedback(new LiteralText("New alias added successfully!"));
+        } catch (Exception e) {
+            sendError(new LiteralText("Uh oh"));
+        }
+        return 0;
+    }
+    private static int listAliases() {
+        HashMap<String, String> aliasMap = getAliases();
+        if (aliasMap == null || aliasMap.isEmpty()) {
+            sendError(new LiteralText("No aliases registered"));
```

Comment:
I also think here, since the user specifically requested to list the aliases and no aliases exist, some kind of feedback is better than throwing an exception.
(But the feedback should be translatable)

Reply:
Agreed. "No aliases registered" is more informative to the user than "An unexpected error occured" which is in this case the alternative

Reply:
Notice that a thrown exception also sends feedback to the client. The only difference is throwing the exception as opposed to sending an error message directly.